internal/repository/advancer.go

```diff
@@ -0,0 +1,196 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/cartesi/rollups-node/internal/node/nodemachine"
+	"github.com/jackc/pgx/v5"
+)
+
+var ErrAdvancerRepository = errors.New("advancer repository error")
+
+type AdvancerRepository struct{ *Database }
+
+func (repository *AdvancerRepository) GetInputs(
+	ctx context.Context,
+	appAddresses []Address,
+) (map[Address][]*Input, error) {
+	result := map[Address][]*Input{}
+	if len(appAddresses) == 0 {
+		return result, nil
+	}
+
+	query := fmt.Sprintf(`
+	SELECT   id, application_address, raw_data
+	FROM     input
+    WHERE    status = 'NONE'
+    AND      application_address IN %s 
+    ORDER BY index ASC, application_address
+    `, toIN(appAddresses)) // NOTE: not sanitized
+	rows, err := repository.db.Query(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("%w (failed querying inputs): %w", ErrAdvancerRepository, err)
+	}
+
+	var input Input
+	scans := []any{&input.Id, &input.AppAddress, &input.RawData}
+	_, err = pgx.ForEachRow(rows, scans, func() error {
+		input := input
+		if _, ok := result[input.AppAddress]; ok { //nolint:gosimple
+			result[input.AppAddress] = append(result[input.AppAddress], &input)
+		} else {
+			result[input.AppAddress] = []*Input{&input}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("%w (failed reading input rows): %w", ErrAdvancerRepository, err)
+	}
+
+	return result, nil
+}
+
+func (repository *AdvancerRepository) StoreResults(
+	ctx context.Context,
+	input *Input,
```

Comment:
```suggestion
	input Input,
```

Reply:
Best to use the pointer than copying the struct.